File: ios/Classes/kits/NavigateKit.swift
//
//  NavigateKit.swift
//  amap_kit
//
//  Created by 杜刚 on 2022/4/25.
//

import Foundation

class NavigateKit: NSObject {
    static let instance = NavigateKit()

    override private init() {}
    
    // 调起高德地图进行导航
    func amapNav(call: FlutterMethodCall) {
        let args = call.arguments as! [String: Any]
        let src = args["src"] as! String
        let lat = args["lat"] as! NSNumber
        let lon = args["lon"] as! NSNumber
        
        var schema = "iosamap://navi?sourceApplication="
        schema += src
        schema += "&lat="
        schema += lat.stringValue
        schema += "&lon="
        schema += lon.stringValue
        schema += "&dev=0"
        
        if #available(iOS 10.0, *) {
            UIApplication.shared.open(URL(string: schema)!)
        } else {
            UIApplication.shared.openURL(URL(string: schema)!)
        }
    }
    
    // 调起百度地图进行导航
    func bmapNav(call: FlutterMethodCall) {
        let args = call.arguments as! [String: Any]
        let location = args["location"] as! String
        let src = args["src"] as! String
        
        var schema = "baidumap://map/navi?location="
        schema += location
        schema += "&src="
        schema += src
        schema += "&coord_type=bd09ll"
        
        if #available(iOS 10.0, *) {
            UIApplication.shared.open(URL(string: schema)!)
        } else {
            UIApplication.shared.openURL(URL(string: schema)!)
        }
    }
}

File: android/src/main/kotlin/com/yuro/amap_kit/kits/NavigateKit.kt
package com.yuro.amap_kit.kits

import android.app.Activity
import android.content.Intent
import android.net.Uri
import io.flutter.plugin.common.MethodCall

object NavigateKit {
    /**
     * 启动高德导航
     */
    fun amapNav(activity: Activity, call: MethodCall) {
        val sb = StringBuffer().apply {
            append("androidamap://navi?sourceApplication=").append(call.argument<String>("src"))
            append("&lat=").append(call.argument<Double>("lat"))
            append("&lon=").append(call.argument<Double>("lon"))
            append("&dev=0")
        }
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(sb.toString())).apply {
            setPackage("com.autonavi.minimap")
        }
        activity.startActivity(intent)
    }

    /**
     * 启动百度导航
     */
    fun bmapNav(activity: Activity, call: MethodCall) {
        val sb = StringBuffer().apply {
            append("baidumap://map/navi?location=").append(call.argument<String>("location"))
            append("&coord_type=bd09ll")
            append("&src=").append(call.argument<String>("src"))
        }
        activity.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(sb.toString())))
    }
}